rrtmg_clouds.md

```diff
@@ -0,0 +1,196 @@
+# RRMTG Clouds with McICA
+
+## Cloud properties
+
+There are three options for the RRTMG `inflag`, as given in the `climt` dictionary:
+
+```python
+rrtmg_cloud_props_dict = {
+    'direct_input': 0,
+    'single_cloud_type': 1,
+    'liquid_and_ice_clouds': 2
+}
+```
+
+With McICA, we cannot use `single_cloud_type`, but can choose between `direct_input` and `liquid_and_ice_clouds`.
+If we choose `direct_input`, we input the `longwave_optical_thickness_due_to_cloud`, `shortwave_optical_thickness_due_to_cloud`, as well as the shortwave parameters `single_scattering_albedo_due_to_cloud`, `cloud_asymmetry_parameter` and `cloud_forward_scattering_fraction`.
+The `cloud_forward_scattering_fraction` is used to scale the other shortwave parameters (`shortwave_optical_thickness_due_to_cloud`, `single_scattering_albedo_due_to_cloud` and `cloud_asymmetry_parameter`), but it is not directly used in the radiative transfer calculations.
+The other cloud properties, namely `cloud_ice_particle_size` and `cloud_water_droplet_radius`, `mass_content_of_cloud_ice_in_atmosphere_layer`, and `mass_content_of_cloud_liquid_water_in_atmosphere_layer` are completely unused.
+The RRTMG `iceflag` and `liqflag` are irrelevant.
+
+On the other hand, if we choose `liquid_and_ice_clouds`,
+any input values for `longwave_optical_thickness_due_to_cloud`, `shortwave_optical_thickness_due_to_cloud`, `single_scattering_albedo_due_to_cloud` and `cloud_asymmetry_parameter` are irrelevant.
+Instead, these parameters are calculated from the cloud ice and water droplet particle sizes (`cloud_ice_particle_size` and `cloud_water_droplet_radius`), as well as the cloud ice and water paths (`mass_content_of_cloud_ice_in_atmosphere_layer`, `mass_content_of_cloud_liquid_water_in_atmosphere_layer`).
+The methods used for the calculations depend on the cloud ice and water properties; `iceflag` and `liqflag` in RRTMG.
+
+
+## Calculation of cloud properties
+
+### Longwave
+
+For the longwave, the only cloud property of interest for calculating radiative fluxes in RRTMG, is the optical depth. This is calculated as follows.
+
+```fortran
```

Comment:
It would be useful to provide a link to the actual file in the code quotations so that people know what is coming from where.

Reply:
Yes, you can!

if you navigate to any file in github like
https://github.com/CliMT/climt/blob/v0.16.0/climt/_lib/rrtmg_lw/parrrtm.f90

if you click on the line number on the left and press shift and click on the another
link, you get a context menu which says get permalink. You should get something like

https://github.com/CliMT/climt/blob/39c1bcacd3b348ec63000d4b57d525e523203883/climt/_lib/rrtmg_lw/parrrtm.f90#L48-L52

It is better to use files from a tagged release (I used the 0.16.0 release above)  so that the link
does not change if the file itself changes.

Reply:
Ok great, so then can I just copy and paste the permalink, or do I need some extra syntax around it in the .rst file?

Reply:
I did some digging, and you can use either of these to show a code block:

```
This code can be found in `this file`_:

.. code-block:: fortran
   :linenos: # optional
   real i, j
   int j(:)

.. _this file: https://github.com/CliMT/climt/blob/39c1bcacd3b348ec63000d4b57d525e523203883/climt/_lib/rrtmg_lw/parrrtm.f90#L48-L52
```
if you are writing copying the code into the rst file and pointing to the file using a github permalink.
http://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-code-block

you could also use
```
.. literalinclude:: ../climt/_lib/rrtmg_lw/parrrtm.f90
    :lines: 48-52
```
to reference a file within the repository. But this has a disadvantage of pointing to the
wrong lines if the file changes (cannot reference an external URL).
http://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-literalinclude

Since you have already added the code into the files, I would go with the first option.

Reply:
just to note, you can see locally check how your file renders by running `make servedocs` in the base directory. You will need to pip install `Sphinx` and `sphinx_rtd_theme` for that make command to work. It should open up the documentation in a local browser.